### PR TITLE
file-store refactor @nsmeds @willnickerson

### PR DESCRIPTION
**Pull request by willnickerson**
Our favorite improvement was the refactoring of our retrieve all function which before this assignment was extremely brittle and verbose due to the fact that it involved multiple array sorts. Our refactoring of this function accounted for the fact that the files would be read and retrieved out of order by assigning each file an index when the get function was called on it. This allowed us to omit all forms of array sorting.

Our fellow 401 student Chris Bruner gave us some tips about storing a frequently used path as a constant, and he noted a couple situations where our code was more verbose than it needed to be.

Andy Beers also reviewed our code and gave us some excellent feedback on small additional features we could add the future. In particular he encouraged us to build logic that ensures that the identifiers for files are unique, which would be extremely important if we were to actually use this package for any kind of large scale file storage.

Mike Burnett, a former Portlander who went to code school and is now a professional JavaScript developer in NYC, gave us some good advice about using ES6 syntax more consistently - using arrow functions, const and let where appropriate, template strings, etc. He gave us some pointers on error handling syntax. He and Chris also pointed out that the README could be more informative.

Getting feedback from other students was extremely valuable as it gave us insights into coding practices and edge cases that we were unaware of or overlooked. It was also great to get outside opinions on the code, especially from someone who isn’t in 401 and has a different perspective.

### CODE REVIEW

File: test/store.test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        done();
+      } else {
+        assert.equal(1,0);
+      }
+    });
+  });
+
+  moreObjects.forEach(obj => {
+    store.save(obj, () => {});
+  });
```
Comment (billyham) [inline test/store.test.js]: You could make test out of this store.save() call. You know that if the callback returns an object then there was an error. The test would be asserting that no object got passed back.
